utils.c

```diff
@@ -0,0 +1,140 @@
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "utils.h"
+
+/* 
+    split_str split s by delim and return a slice containing all subs-trings of s after the split 
+    Notice: caller is responsible to free the memmory of the return value
+*/
+char** split_str(char *s, const char *delim) {
+
+    // TODO - fix case of trailing delim
+
+
+    // copy s (we will be using strtoken which modifies the original string)
+    char *s_dup = _strdup(s);
+    if (s_dup == NULL) {
+        printf("ERROR: failed to allocate memmory");
+        return NULL; // nothing to clean
+    }
+
+    // using tmp to count appearances of delim (we will be using s_dup again)
+    char *tmp = s_dup;
+    int count = 0;
+    // count appearacnes of delim
+    while (*tmp) {
+        if (*tmp == *delim) {
```

Comment:
Either only accept char delim, or support delimiters that are bigger than a single character.